twine/utils.py

```diff
@@ -99,19 +101,44 @@ def get_config(path: str = "~/.pypirc") -> Dict[str, RepositoryConfig]:
     return dict(config)
 
 
+def validate_url(repository_url: Optional[str]) -> Any:
+    """Validate the given url for allowed schemes and components"""
+
+    if not repository_url:
+        return None
+
+    # Scheme should always be https, and the url should at minimum
+    # contain scheme and host
+    validator = (
+        validators.Validator()
+        .allow_schemes("https")
+        .require_presence_of("scheme", "host")
+    )
+
+    # Only return scheme when the url is valid
+    url = uri_reference(repository_url)
+    try:
+        validator.validate(url)
+    except Exception:
+        return None
+
+    return url.scheme
```

Comment:
Why are we returning the scheme here? This doesn't appear to have any correlation to the rest of the function

Reply:
I was returning scheme here since we check if the scheme is present at https://github.com/pypa/twine/pull/602/files#diff-547bab308763f89cacec226151fcbb80R134 . I guess we can return boolean from this function and check `if scheme` and `if not scheme` in the subsequent code after this

Reply:
I have updated `validate_url` to return `bool` (`True` when the url is validated correctly, and `False` otherwise) to be more consistent.